tests/run-pass/intptrcast-truncate.rs

```diff
@@ -0,0 +1,8 @@
+// compile-flags: -Zmiri-seed=0000000000000000
+
+fn main() {
+    let x = &42 as *const i32 as usize; 
+    let y = x * 2;
```

Comment:
Can we also assert something about `x`? Like, `assert_eq!(y+y, x)`?

Reply:
This breaks the test:
```rust
error[E0080]: Miri evaluation error: a raw memory access tried to access part of a pointer value as raw bytes
  --> /Users/christian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/src/libcore/fmt/num.rs:30:35
   |
30 |         fn to_u64(&self) -> u64 { *self as u64 }
   |                                   ^^^^^^^^^^^^ Miri evaluation error: a raw memory access tried to access part of a pointer value as raw bytes
   |
```